pkg/proxy/winkernel/proxier.go

```diff
@@ -527,10 +538,11 @@ func NewProxier(
 	}
 
 	hnsNetworkInfo, err := hns.getNetworkByName(hnsNetworkName)
-	if err != nil {
-		klog.Errorf("Unable to find Hns Network specified by %s. Please check environment variable KUBE_NETWORK or network-name flag", hnsNetworkName)
-		return nil, err
+	for err != nil {
+		klog.Errorf("Unable to find Hns Network specified by %s. Please check network name and CNI deployment", hnsNetworkName)
+		hnsNetworkInfo, err = hns.getNetworkByName(hnsNetworkName)
```

Comment:
Probably you want to try to add some sleep here, rather than being very aggressive ?